app.go

```diff
@@ -14,15 +14,13 @@ import (
 // Without an App you can't do much!
 type App struct {
 	Options
-	// Middleware returns the current MiddlewareStack for the App/Group.
-	Middleware    *MiddlewareStack `json:"-"`
-	ErrorHandlers ErrorHandlers    `json:"-"`
-	router        *mux.Router
-	moot          *sync.RWMutex
-	routes        RouteList
-	root          *App
-	children      []*App
-	filepaths     []string
+	// Middleware, ErrorHandlers, router, and filepaths are moved to Home.
+	Home
```

Comment:
This is a breaking change, should we add fixers to update existing code?

Reply:
could you please provide some more details about what is broken by this?
Actually, this patch (#2226) fixed the breaking change introduced by #2185. (more details in #2222)

Reply:
Any code that accesses `App.Middleware` or `App.Errorhandlers` will now fail to compile when upgrading from `v0.18.5` to `v0.18.6`

Reply:
Oh, really, I will check. I think they are still accessible since they are moved to `Home` which is embedded by `App`.

Reply:
With my testing, it still works fine and I think there is no reason accessing `App.Middleware` is broken. See [1]

However, it seems like my buffalo binary seems like broken. `buffalo routes` and `buffalo task middleware` are not working with error:

```
.grifter/main.go:9:2: package grifts is not in GOROOT (/opt/google/go/src/grifts)
```

[1] https://github.com/gobuffalo/buffalo/pull/2239/files#diff-5b9f0b55bb5e0ae16b6fde53332a8923b6d445a022da239a5d23e66e7c7082e5R35

Reply:
Works fine as below. The issue with `buffalo routes` and `buffalo task` seems related to `buffalo new app --skip-pop --skip-webpack --api`. With the basic webapp, no issue with them. (Will look at that issue later. It seems like github is not working properly now)

```console
$ buffalo tasks middleware
-> /
	github.com/gobuffalo/mw-csrf.glob..func1
	github.com/gobuffalo/buffalo-pop/v3/pop/popmw.Transaction.func2
	github.com/gobuffalo/mw-i18n/v2.*Translator.Middleware.func1
$ vi go.mod
$ go mod tidy
$ buffalo tasks middleware
-> http://127.0.0.1:3000/
	github.com/gobuffalo/mw-csrf.glob..func1
	github.com/gobuffalo/buffalo-pop/v3/pop/popmw.Transaction.func2
	github.com/gobuffalo/mw-i18n/v2.*Translator.Middleware.func1
$ git diff actions
diff --git a/actions/app.go b/actions/app.go
index be86b4c..3552ab2 100644
--- a/actions/app.go
+++ b/actions/app.go
@@ -52,6 +52,8 @@ func App() *buffalo.App {
                // Log request parameters (filters apply).
                app.Use(paramlogger.ParameterLogger)
 
+               app.Middleware.Clear()
+
                // Protect against CSRF attacks. https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)
                // Remove to disable this.
                app.Use(csrf.New)
$ 
```

Reply:
I assumed embedding home would cause a similar issue like the original problem, but thanks for double checking.